flang/lib/Optimizer/Transforms/AffineDemotion.cpp

```diff
@@ -32,43 +42,45 @@ using namespace fir;
 
 namespace {
 
-class AffineLoadConversion : public OpRewritePattern<mlir::AffineLoadOp> {
+class AffineLoadConversion : public OpConversionPattern<mlir::AffineLoadOp> {
 public:
-  using OpRewritePattern<mlir::AffineLoadOp>::OpRewritePattern;
+  using OpConversionPattern<mlir::AffineLoadOp>::OpConversionPattern;
 
-  LogicalResult matchAndRewrite(mlir::AffineLoadOp op,
-                                PatternRewriter &rewriter) const override {
-    SmallVector<Value> indices(op.getMapOperands());
+  LogicalResult
+  matchAndRewrite(mlir::AffineLoadOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    SmallVector<Value> indices(adaptor.indices());
```

Comment:
nit: inconsistent mix of explicit namespaces and implied namespaces in this file.